src/main/c/netty_jni_util.c

```diff
@@ -461,7 +437,6 @@ jint netty_jni_util_JNI_OnLoad(JavaVM* vm, void* reserved, const char* libname,
 #endif /* NETTY_JNI_UTIL_BUILD_STATIC */
 
     jint ret = load_function(env, packagePrefix);
-    free(packagePrefix);
```

Comment:
as explained in the commit descriptions, the goal is to avoid strdup in the load_function. the reason why strdup was called in the first place was because the packagePrefix is invalid after load_function returns. we can avoid this by simply passing the ownership of packagePrefix to load_function. if we free packagePrefix then it will be a UAF.